operator/src/main/java/org/bf2/operator/operands/KafkaInstanceConfigurations.java

```diff
@@ -77,12 +87,15 @@ private void overlayWithConfigProperties(Map<String, String> defaultValues, Unar
     }
 
     public KafkaInstanceConfiguration getConfig(InstanceType config) {
-        return configs.get(config);
+        return configs.get(config.lowerName);
     }
 
     public KafkaInstanceConfiguration getConfig(ManagedKafka managedKafka) {
-        // TDB where to get name from
-        return configs.get(InstanceType.STANDARD);
+        return configs.get(getInstanceType(managedKafka));
+    }
+
+    public static String getInstanceType(ManagedKafka managedKafka) {
+        return OperandUtils.getOrDefault(managedKafka.getMetadata().getLabels(), PROFILE_TYPE, InstanceType.STANDARD.lowerName);
```

Comment:
It might be useful for development to allow a config property to override the instance type. For example
```
mvn quarkus:dev -Dmanagedkafka.instancetype-override=DEVELOPER
```

Reply:
I'm not sure when we would use that.  If we're already in the dev profile we're not creating real instances.  You can also just declaratively update the managedkafka to be a developer instance.

Reply:
> If we're already in the dev profile we're not creating real instances.

What do you mean by that? I frequently run it in dev mode to create/update instances in OSD. Prior to profiles I would override the resource requests manually, but setting this to `DEVELOPER` now that there are profiles is much more convenient.

Reply:
> What do you mean by that?

In quarkus:dev the dev operands are used - see the package org.bf2.operator.operands.dev - these are not the same as standard or developer instances.

Reply:
> but they're close enough for a lot of scenarios

Sure it's primarily for minikube.  It's unfortunate that we have yet another overloaded term :(  To recap - in dev mode you won't even hit this logic in KafkaCluster related to profiles and configuration overrides.  You just get the dev instance.  If for whatever reason you want the managedkafka to have the profile label, it can be there, but won't get used or propagated.